### PR TITLE
Added activity pie charts

### PR DESCRIPTION
**Pull request by aidan-epperly**
This PR adds activity pie charts to break down the percent of the activity that the popular repos are responsible for.
image

### CODE REVIEW

**Comment by aidan-epperly** [comment]
Found a summation error, let me fix that real quick

**Comment by IanLee1521** [comment]
This is really cool, and a great idea.

I wonder if it makes sense to include the the top N separately in the pie. E.g. rather than just the "top 10" or "not top 10" have "#1", "#2", ..., "#10", "non top 10"

File: js/explore/largeRepos/pie_activityLines.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    var url = ghDataDir + '/labRepos_ActivityLines.json';
+    var files = [url];
+    Promise.all(files.map(url => d3.json(url))).then(response => {
+            var data = reformatData(response[0]);
+            drawGraph(data, areaID);
+        });
+
+    // Draw graph from data
+    function drawGraph(data, areaID) {
+        var graphHeader = 'Lines Added This Year';
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityLines.js:13]: The title for "Lines Added" seems a bit misleading as far as new lines vs edited lines....
Perhaps "Line Changes" would be a good alternative? (I'm having a hard time coming up with another good option that fits without shrinking text, unless we want to leave out "This Year".)

```suggestion
        var graphHeader = 'Line Changes This Year';
```

Comment (aidan-epperly) [reply js/explore/largeRepos/pie_activityLines.js:13]: To me, changes would denote deletions and additions, though I get what you mean about not being able to fit a better title in the space. We could add "this year" to the tip and take it out in the title?

Comment (aidan-epperly) [reply js/explore/largeRepos/pie_activityLines.js:13]: Something like, "Line edits and additions"

File: js/explore/largeRepos/pie_activityCommits.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        var tip = d3
+            .tip()
+            .attr('class', 'd3-tip')
+            .offset(function() {
+                return [this.getBBox().height / 2, 0];
+            })
+            .html(function(d) {
+                var units = ' Repos';
+                if (d.data.count == 1) {
+                    units = ' Repo';
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityCommits.js:42]:
```suggestion
                var units = ' Commits';
                if (d.data.count == 1) {
                    units = ' Commit';
```

File: js/explore/largeRepos/pie_activityCommits.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            .attr('class', 'graphtitle bignum')
+            .attr('x', 0)
+            .attr('y', -25)
+            .attr('text-anchor', 'middle')
+            .text(dataTotalCount);
+    }
+
+    // Turn json obj into desired working data
+    function reformatData(obj) {
+        var data = [{ label: 'Commits by Top Ten Repos', count: totalCommits(obj, mostPopularRepositories.map(d => `${d.owner}/${d.name}`)) }, { label: 'Commits by Other Repos', count: totalCommits(obj) }];
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityCommits.js:135]:
```suggestion
        var data = [{ label: 'In Top Ten LLNL Repos', count: totalCommits(obj, mostPopularRepositories.map(d => `${d.owner}/${d.name}`)) }, { label: 'In Other LLNL Repos', count: totalCommits(obj) }];
```

File: js/explore/largeRepos/pie_activityLines.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityLines.js:13]:
```suggestion
        var graphHeader = 'Line Edits and Additions';
```

File: js/explore/largeRepos/pie_activityLines.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        var tip = d3
+            .tip()
+            .attr('class', 'd3-tip')
+            .offset(function() {
+                return [this.getBBox().height / 2, 0];
+            })
+            .html(function(d) {
+                var units = ' Repos';
+                if (d.data.count == 1) {
+                    units = ' Repo';
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityLines.js:42]:
```suggestion
                var units = ' Lines This Year';
                if (d.data.count == 1) {
                    units = ' Line This Year';
```

File: js/explore/largeRepos/pie_activityLines.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            .attr('class', 'graphtitle bignum')
+            .attr('x', 0)
+            .attr('y', -25)
+            .attr('text-anchor', 'middle')
+            .text(d3.format('.4s')(dataTotalCount));
+    }
+
+    // Turn json obj into desired working data
+    function reformatData(obj) {
+        var data = [{ label: 'Lines by Top Ten Repos', count: totalCommits(obj, mostPopularRepositories.map(d => `${d.owner}/${d.name}`)) }, { label: 'Lines by Other Repos', count: totalCommits(obj) }];
```
Comment (LRWeber) [inline js/explore/largeRepos/pie_activityLines.js:135]:
```suggestion
        var data = [{ label: 'In Top Ten LLNL Repos', count: totalCommits(obj, mostPopularRepositories.map(d => `${d.owner}/${d.name}`)) }, { label: 'In Other LLNL Repos', count: totalCommits(obj) }];
```

**Comment by aidan-epperly** [comment]
image

**Pull request merged**